### PR TITLE
Zj/more javascript landing page

### PR DESCRIPTION
**Pull request by zacjones93**
This PR adds the layout for the JavaScript landing page:

CleanShot 2021-01-29 at 16 44 28@2x

dog i cant handle this - WZP3qaxYj10gU

### CODE REVIEW

**Comment by vojtaholik** [comment]
the whole "Composing Closures and Callbacks in JavaScript" section has a darker background, which is meant to group things under one topic visually. (see state management on react page) But it looks like Debugging and favorites don't quite belong under it.

**Comment by MaggieAppleton** [comment]
Yeah I think this page needs a different layout to the React one based around the content. I'm going to layout some ideas in Figma then rearrange

**Comment by zacjones93** [comment]
That makes sense. I used the React page components to get the data up and running with as little friction as possible so hopefully rearranging the layout won't be an issue once we've landed on a solution

**Comment by joelhooks** [comment]
I added the `RFC` tag, which we use as a "request for comments" - it also blocks the auto-merge robot so should be removed when this is ready

**Comment by zacjones93** [comment]
I added the async card which looks great on desktop but can't figure out why the styling gets cramped on mobile

## Desktop
In-Depth Up-to-Date JavaScript Tutorials for 2021  egghead io

## Mobile
In-Depth Up-to-Date JavaScript Tutorials for 2021  egghead io

File: src/components/search/curated/javascript/index.tsx [hunk 1]
```diff
@@ -108,21 +109,32 @@ When someone comes to you and asks "Hey! Can you build this?", you'll be able to
...
         </Card>
-        <Card
-          className="md:col-span-8"
-          resource={composingCallbacks}
-          location={location}
-        >
-          <Collection />
-        </Card>
+        <div className="md:col-span-8 min-w-full gap-5">
+          <div className="md:grid md:grid-cols-8 flex flex-col w-max items-start gap-5">
```
Comment (zacjones93) [inline src/components/search/curated/javascript/index.tsx:121]: It has something to do with how I'm applying the nested grid here

**Comment by zacjones93** [comment]
slight margin issue at the bottom but other than that this is gtg
In-Depth Up-to-Date JavaScript Tutorials for 2021  egghead io

**Review by laurosilvacom — APPROVED** [review]

**Pull request merged**